src/NUnitTestAdapter/NavigationDataProvider.cs

```diff
@@ -0,0 +1,150 @@
+﻿// ****************************************************************
+// Copyright (c) 2016 NUnit Software. All rights reserved.
+// ****************************************************************
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+using Mono.Cecil.Rocks;
+
+namespace NUnit.VisualStudio.TestAdapter
+{
+    public class NavigationDataProvider : MarshalByRefObject
```

Comment:
It looks like you removed all of the AppDomains. Is there still one in the unchanged code that makes this need to be marshalable? I thought the advantage of Cecil for us in the Engine was that we didn't need AppDomains.